app/src/main/java/com/firebase/uidemo/BaseActivity.java

```diff
@@ -0,0 +1,16 @@
+package com.firebase.uidemo;
+
+import android.os.Bundle;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.view.WindowCompat;
+
+public class BaseActivity extends AppCompatActivity {
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        
+        // Enable edge-to-edge for all activities
+        WindowCompat.setDecorFitsSystemWindows(getWindow(), false);
```

Comment:
Given that this is just calling a single method, I'd prefer that we call this method in every other Activity rather than creating this new `BaseActivity` that is extended by all of them.

When we migrate the sample app to Kotlin, let's try to get rid of all of these "Base..." classes.